src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -191,10 +193,7 @@ public boolean makeOTPRequestAndUpdateMatchingItinerary() {
                 // If the updated trip status is upcoming and the end time of the current matching itinerary is in the
                 // past, this means the trip has completed and the next possible time the trip occurs should be
                 // calculated
-                if (
-                    journeyState.tripStatus == TripStatus.TRIP_UPCOMING &&
-                        matchingItinerary.endTime.before(DateTimeUtils.nowAsDate())
-                ) {
+                if (journeyState.tripStatus == TripStatus.TRIP_UPCOMING && !matchingItinerary.hasEnded()) {
```

Comment:
Shouldn't this be `matchingItinerary.hasEnded()` (without the "not").

Reply:
Yep, fixed in d95910c71758df3212e5f878a91a509f02839dd0.